### PR TITLE
Implement role assignment and removal APIs

### PR DESCRIPTION
**Pull request by JAORMX**
This implements the role assignment and removal in the Minder server.
This already takes into account the project hierarchy.

### CODE REVIEW

File: internal/controlplane/handlers_authz.go [hunk 1]
```diff
@@ -265,12 +265,121 @@ func (*Server) ListRoleAssignments(
...
+
+	authzrole, err := authz.ParseRole(role)
+	if err != nil {
+		return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
+	}
+
+	// Determine target project.
+	entityCtx := engine.EntityFromContext(ctx)
+	currentProjectID := entityCtx.Project.ID
+	targetProjectID := currentProjectID
```
Comment (eleftherias) [inline internal/controlplane/handlers_authz.go:287]: I'm wondering if there's any benefit in differentiating the current project and the target project. 
If we have a command like: `minder role assign --subject user1 --project p1`, then the project in the context (i.e current project) would also be `p1`. At least that's how we do it for other commands.

Comment (JAORMX) [reply internal/controlplane/handlers_authz.go:287]: I think you're right. Let me remove this logic from here.

Comment (JAORMX) [reply internal/controlplane/handlers_authz.go]: done

**Review by eleftherias — CHANGES_REQUESTED** [review]

File: internal/controlplane/handlers_authz.go [hunk 2]
```diff
@@ -265,12 +267,92 @@ func (*Server) ListRoleAssignments(
...
+	if _, err := s.store.GetUserBySubject(ctx, sub); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, util.UserVisibleError(codes.NotFound, "User not found")
+		}
+		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
+	}
+
+	// Determine target project.
+	entityCtx := engine.EntityFromContext(ctx)
+	targetProjectID := entityCtx.Project.ID
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply internal/controlplane/handlers_authz.go:297]: I can add this validation in a separate PR (should probably merge before this).

File: internal/controlplane/handlers_authz.go [hunk 3]
```diff
@@ -265,12 +267,92 @@ func (*Server) ListRoleAssignments(
...
+	// Verify if user exists
+	if _, err := s.store.GetUserBySubject(ctx, sub); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, util.UserVisibleError(codes.NotFound, "User not found")
+		}
+		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
+	}
+
+	// Determine target project.
+	entityCtx := engine.EntityFromContext(ctx)
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply internal/controlplane/handlers_authz.go:296]: done.

File: internal/controlplane/handlers_authz.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (JAORMX) [reply internal/controlplane/handlers_authz.go:297]: Actually... we might not need that as the hierarchy is already checked in OpenFGA

[the remark this answers is not in the document]
Comment (evankanderson) [reply internal/controlplane/handlers_authz.go:297]: Also, `targetProjectID` feels a little funny, as there should only be one `projectID` in this context.  There are two users (the _caller_ and the _subject_ user who's being granted permissions), so those names make sense.

File: internal/controlplane/handlers_authz.go [hunk 4]
```diff
@@ -265,12 +267,88 @@ func (*Server) ListRoleAssignments(
...
+	targetProjectID := entityCtx.Project.ID
+
+	if err := s.authzClient.Delete(ctx, sub, authzrole, targetProjectID); err != nil {
+		return nil, status.Errorf(codes.Internal, "error writing role assignment: %v", err)
+	}
```
Comment (evankanderson) [inline internal/controlplane/handlers_authz.go:344]: Is it possible for `Delete` to return a "Not Found"?  If so, do we want to handle it specially here?

File: proto/minder/v1/minder.proto [hunk 5]
```diff
@@ -1511,6 +1511,6 @@ message RoleAssignment {
     string role = 1;
     // subject is the subject to which the role is assigned.
     string subject = 2;
-    // context is the context in which the role is assigned.
-    optional Context context = 3;
+    // project is the project in which the role is assigned.
+    optional string project = 3;
```
Comment (evankanderson) [inline proto/minder/v1/minder.proto:1515]: Just use tag 4.  There's no prize for keeping consecutive tag numbers.

**Review by evankanderson — DISMISSED** [review]

File: internal/controlplane/handlers_authz.go (hunk 4 above)
Comment (JAORMX) [reply internal/controlplane/handlers_authz.go]: no, I special cased that in the authz wrapper.

**Review by evankanderson — APPROVED** [review]

**Review by eleftherias — APPROVED** [review]

**Pull request merged**